From 07cafa6c8de44be206f591d28ffc0bd463013ffa Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 4 Jun 2013 11:25:56 +0900
Subject: [PATCH 671/675] LTSI: ASoC: rcar: tidyup framework mismatch

Original Kernel is based on latest framework,
but it was not back-ported yet.
tidyup it.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
---
 sound/soc/sh/rcar/core.c | 11 +++--------
 sound/soc/sh/rcar/gen.c  |  6 +++---
 2 files changed, 6 insertions(+), 11 deletions(-)

diff --git a/sound/soc/sh/rcar/core.c b/sound/soc/sh/rcar/core.c
index a357060..e73e0ac 100644
--- a/sound/soc/sh/rcar/core.c
+++ b/sound/soc/sh/rcar/core.c
@@ -743,10 +743,6 @@ static struct snd_soc_platform_driver rsnd_soc_platform = {
 	.pcm_free	= rsnd_pcm_free,
 };
 
-static const struct snd_soc_component_driver rsnd_soc_component = {
-	.name		= "rsnd",
-};
-
 /*
  *	rsnd probe
  */
@@ -799,6 +795,8 @@ static int rsnd_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return ret;
 
+	dev_set_drvdata(dev, priv);
+
 	/*
 	 *	asoc register
 	 */
@@ -808,15 +806,12 @@ static int rsnd_probe(struct platform_device *pdev)
 		return ret;
 	}
 
-	ret = snd_soc_register_component(dev, &rsnd_soc_component,
-					 priv->daidrv, rsnd_dai_nr(priv));
+	ret = snd_soc_register_dais(dev, priv->daidrv, rsnd_dai_nr(priv));
 	if (ret < 0) {
 		dev_err(dev, "cannot snd dai register\n");
 		goto exit_snd_soc;
 	}
 
-	dev_set_drvdata(dev, priv);
-
 	pm_runtime_enable(dev);
 
 	dev_info(dev, "probed\n");
diff --git a/sound/soc/sh/rcar/gen.c b/sound/soc/sh/rcar/gen.c
index 80ef4c8..59a1aa8 100644
--- a/sound/soc/sh/rcar/gen.c
+++ b/sound/soc/sh/rcar/gen.c
@@ -163,9 +163,9 @@ static int rsnd_gen1_probe(struct platform_device *pdev,
 
 	gen->ops = &rsnd_gen1_ops;
 
-	gen->base[RSND_GEN1_SRU] = devm_ioremap_resource(dev, sru_res);
-	gen->base[RSND_GEN1_ADG] = devm_ioremap_resource(dev, adg_res);
-	gen->base[RSND_GEN1_SSI] = devm_ioremap_resource(dev, ssi_res);
+	gen->base[RSND_GEN1_SRU] = ioremap_nocache(sru_res->start, resource_size(sru_res));
+	gen->base[RSND_GEN1_ADG] = ioremap_nocache(adg_res->start, resource_size(adg_res));
+	gen->base[RSND_GEN1_SSI] = ioremap_nocache(ssi_res->start, resource_size(ssi_res));
 	if (!gen->base[RSND_GEN1_SRU] ||
 	    !gen->base[RSND_GEN1_ADG] ||
 	    !gen->base[RSND_GEN1_SSI]) {
-- 
1.8.2

